### PR TITLE
Updating PUT operation requirements for SourceControls and SourceControlSyncJobs, updating streams’ value property, and updating examples. 

### PR DESCRIPTION
**Pull request by Francisco-Gamino**
The PR contains the following changes:
- Updating source control security token requirements for PUT operation.
- Fixing SourceControlSyncJobs examples by replacing property 'syncJobId' for 'sourceControlSyncJobId'.
- Making commitId property required and allowing empty values. 
- Making stream's value property an object instead of a string, and updating examples.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/automation/resource-manager/Microsoft.Automation/preview/2017-05-15-preview/sourceControl.json [hunk 1]
```diff
@@ -493,8 +494,8 @@
         },
         "securityToken": {
           "type": "string",
-          "maxLength": 1024,
-          "description": "Gets or sets the authorization token for the repo of the source control."
+          "description": "Gets or sets the authorization token for the repo of the source control.",
```
Comment (annatisch) [inline specification/automation/resource-manager/Microsoft.Automation/preview/2017-05-15-preview/sourceControl.json]: Please avoid terminology like "Gets or sets" as this isn't idiomatic for all generated SDK languages. Just stick with the description. This also applies to the other descriptions below.

**Review by annatisch — COMMENTED** [review]
Looks good - just a new minor things.
Just to confirm - are you aware that these changes are breaking and will result in a major version bump for all generated SDKs?

File: specification/automation/resource-manager/Microsoft.Automation/preview/2017-05-15-preview/sourceControl.json [hunk 2]
```diff
@@ -493,8 +494,8 @@
         },
         "securityToken": {
           "type": "string",
```
[the remark this answers is not in the document]
Comment (Francisco-Gamino) [reply specification/automation/resource-manager/Microsoft.Automation/preview/2017-05-15-preview/sourceControl.json]: I missed that, thanks. Fixed.

File: specification/automation/resource-manager/Microsoft.Automation/preview/2017-05-15-preview/sourceControl.json [hunk 3]
```diff
@@ -352,7 +352,8 @@
...
```
[the remark this answers is not in the document]
Comment (Francisco-Gamino) [reply specification/automation/resource-manager/Microsoft.Automation/preview/2017-05-15-preview/sourceControl.json]: Fixed.

File: specification/automation/resource-manager/Microsoft.Automation/preview/2017-05-15-preview/sourceControl.json [hunk 4]
```diff
@@ -352,7 +352,8 @@
...
-          "description": "Gets or sets the authorization token for the repo of the source control."
+          "description": "Gets or sets the authorization token for the repo of the source control.",
```
[the remark this answers is not in the document]
Comment (Francisco-Gamino) [reply specification/automation/resource-manager/Microsoft.Automation/preview/2017-05-15-preview/sourceControl.json]: Thank you for the suggestion. Fixed.

File: specification/automation/resource-manager/Microsoft.Automation/preview/2017-05-15-preview/sourceControl.json (hunk 1 above)
Comment (Francisco-Gamino) [reply specification/automation/resource-manager/Microsoft.Automation/preview/2017-05-15-preview/sourceControl.json]: Fixed.

**Comment by Francisco-Gamino** [comment]
@annatisch: Thank you for the fast review. I've updated the specs, could you please take another look? 

Regarding your comment: "Just to confirm - are you aware that these changes are breaking and will result in a major version bump for all generated SDKs?"
@vrdmr: Do you seen any issues with a major version bump for all regenerated Azure Automation SDKs?

Thanks!
Francisco

**Comment by annatisch** [comment]
Thanks @Francisco-Gamino for the quick response!

Whether you publish breaking changes to the SDKs is up to you - we generally recommend that service teams try to group as many breaking changes together so as to avoid frequently breaking releases - however you best know your customers and the frequency of major changes to your API :)

Once the CI passes I will take another look!

**Review by annatisch — APPROVED** [review]
Looks good to me! And thanks for fixing up the other descriptions :)

**Comment by Francisco-Gamino** [comment]
Awesome! Thanks annatisch!

**Pull request merged**